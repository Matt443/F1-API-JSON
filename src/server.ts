import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}