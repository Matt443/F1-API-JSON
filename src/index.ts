import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
