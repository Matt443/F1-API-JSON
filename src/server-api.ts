import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
