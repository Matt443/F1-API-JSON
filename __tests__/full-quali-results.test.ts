import { getFullQualiResults } from "../src/server";

test("get full quali results", async () => {
    expect(await getFullQualiResults(2008, "Great-Britain")).toMatchSnapshot();
});
