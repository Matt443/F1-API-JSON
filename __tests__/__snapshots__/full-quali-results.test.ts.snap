// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get full quali results 1`] = `
[
  {
    "code": "KOV",
    "laps": 15,
    "name": "Heikki Kovalainen",
    "number": 23,
    "position": "1",
    "team": "McLaren Mercedes",
    "times": {
      "q1": "1:19.957",
      "q2": "1:19.597",
      "q3": "1:21.049",
    },
  },
  {
    "code": "WEB",
    "laps": 16,
    "name": "Mark Webber",
    "number": 10,
    "position": "2",
    "team": "Red Bull Renault",
    "times": {
      "q1": "1:20.982",
      "q2": "1:19.710",
      "q3": "1:21.554",
    },
  },
  {
    "code": "RAI",
    "laps": 18,
    "name": "Kimi Räikkönen",
    "number": 1,
    "position": "3",
    "team": "Ferrari",
    "times": {
      "q1": "1:20.370",
      "q2": "1:19.971",
      "q3": "1:21.706",
    },
  },
  {
    "code": "HAM",
    "laps": 13,
    "name": "Lewis Hamilton",
    "number": 22,
    "position": "4",
    "team": "McLaren Mercedes",
    "times": {
      "q1": "1:20.288",
      "q2": "1:19.537",
      "q3": "1:21.835",
    },
  },
  {
    "code": "HEI",
    "laps": 16,
    "name": "Nick Heidfeld",
    "number": 3,
    "position": "5",
    "team": "Sauber BMW",
    "times": {
      "q1": "1:21.022",
      "q2": "1:19.802",
      "q3": "1:21.873",
    },
  },
  {
    "code": "ALO",
    "laps": 15,
    "name": "Fernando Alonso",
    "number": 5,
    "position": "6",
    "team": "Renault",
    "times": {
      "q1": "1:20.998",
      "q2": "1:19.992",
      "q3": "1:22.029",
    },
  },
  {
    "code": "PIQ",
    "laps": 19,
    "name": "Nelson Piquet",
    "number": 6,
    "position": "7",
    "team": "Renault",
    "times": {
      "q1": "1:20.818",
      "q2": "1:20.115",
      "q3": "1:22.491",
    },
  },
  {
    "code": "VET",
    "laps": 16,
    "name": "Sebastian Vettel",
    "number": 15,
    "position": "8",
    "team": "STR Ferrari",
    "times": {
      "q1": "1:20.318",
      "q2": "1:20.109",
      "q3": "1:23.251",
    },
  },
  {
    "code": "MAS",
    "laps": 16,
    "name": "Felipe Massa",
    "number": 2,
    "position": "9",
    "team": "Ferrari",
    "times": {
      "q1": "1:20.676",
      "q2": "1:20.086",
      "q3": "1:23.305",
    },
  },
  {
    "code": "KUB",
    "laps": 12,
    "name": "Robert Kubica",
    "number": 4,
    "position": "10",
    "team": "Sauber BMW",
    "times": {
      "q1": "1:20.444",
      "q2": "1:19.788",
      "q3": "DNF",
    },
  },
  {
    "code": "COU",
    "laps": 14,
    "name": "David Coulthard",
    "number": 9,
    "position": "11",
    "team": "Red Bull Renault",
    "times": {
      "q1": "1:21.224",
      "q2": "1:20.174",
    },
  },
  {
    "code": "GLO",
    "laps": 14,
    "name": "Timo Glock",
    "number": 12,
    "position": "12",
    "team": "Toyota",
    "times": {
      "q1": "1:20.893",
      "q2": "1:20.274",
    },
  },
  {
    "code": "BOU",
    "laps": 16,
    "name": "Sebastien Bourdais",
    "number": 14,
    "position": "13",
    "team": "STR Ferrari",
    "times": {
      "q1": "1:20.584",
      "q2": "1:20.531",
    },
  },
  {
    "code": "TRU",
    "laps": 13,
    "name": "Jarno Trulli",
    "number": 11,
    "position": "14",
    "team": "Toyota",
    "times": {
      "q1": "1:21.145",
      "q2": "1:20.601",
    },
  },
  {
    "code": "NAK",
    "laps": 14,
    "name": "Kazuki Nakajima",
    "number": 8,
    "position": "15",
    "team": "Williams Toyota",
    "times": {
      "q1": "1:21.407",
      "q2": "1:21.112",
    },
  },
  {
    "code": "BAR",
    "laps": 9,
    "name": "Rubens Barrichello",
    "number": 17,
    "position": "16",
    "team": "Honda",
    "times": {
      "q1": "1:21.512",
    },
  },
  {
    "code": "BUT",
    "laps": 11,
    "name": "Jenson Button",
    "number": 16,
    "position": "17",
    "team": "Honda",
    "times": {
      "q1": "1:21.631",
    },
  },
  {
    "code": "ROS",
    "laps": 6,
    "name": "Nico Rosberg",
    "number": 7,
    "position": "18",
    "team": "Williams Toyota",
    "times": {
      "q1": "1:21.668",
    },
  },
  {
    "code": "SUT",
    "laps": 9,
    "name": "Adrian Sutil",
    "number": 20,
    "position": "19",
    "team": "Force India Ferrari",
    "times": {
      "q1": "1:21.786",
    },
  },
  {
    "code": "FIS",
    "laps": 8,
    "name": "Giancarlo Fisichella",
    "number": 21,
    "position": "20",
    "team": "Force India Ferrari",
    "times": {
      "q1": "1:21.885",
    },
  },
]
`;
